src/runtime/hexagon_remote/sim_host.cpp

```diff
@@ -109,19 +109,30 @@ int write_memory(int dest, const void *src, int size) {
         // WriteMemory only works with powers of 2, so align down. It
         // also only writes up to 8 bytes, so we need to do this
         // repeatedly until we've finished copying the buffer.
-        int next = 1;
-        if (size >= 8) next = 8;
-        else if (size >= 4) next = 4;
-        else if (size >= 2) next = 2;
-        HEXAPI_Status status = sim->WriteMemory(dest, next, *reinterpret_cast<const HEX_8u_t*>(src));
+        HEX_8u_t src_next;
```

Comment:
I think this code would be easier to read if src_next was named src_chunk. I believe it holds the actual bits, not a pointer to the bits. src_next implies the next value of src and src is a pointer. I'd also name "size" to "remaining_size" and "size_next" to "chunk_size". (To me size_next implies it is the size to use on the next iteration of the loop.) Other naming is possible, and this is surely not a major issue.